connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/WorkerCoordinator.java

```diff
@@ -211,7 +211,13 @@ protected void onJoinComplete(int generation, String memberId, String protocol,
     }
 
     @Override
-    protected Map<String, ByteBuffer> performAssignment(String leaderId, String protocol, List<JoinGroupResponseMember> allMemberMetadata) {
+    protected Map<String, ByteBuffer> onLeaderElected(String leaderId,
+                                                      String protocol,
+                                                      List<JoinGroupResponseMember> allMemberMetadata,
+                                                      boolean skipAssignment) {
+        if (skipAssignment)
+            throw new IllegalStateException("Can't skip assignment because Connect does not support static membership.");
```

Comment:
Would it be safer to just log an error and ignore?  That would make a difference if we say have a bug and return skipAssignment=true by mistake.

Reply:
Hmm, is it safe to ignore? What is the behavior if the client provides the new assignment anyway when `skipAssignment` is set? It might be worse to let connect generate a new assignment and let the coordinator ignore it. I think I prefer to fail fast.

For what it's worth, I don't think we have a great support for test-only assertions today.

Reply:
Ok, maybe I misunderstood the intent of `skipAssignment`, my understanding was that was intended as an optimization on the client (so that the static members wouldn't do assignment on taking leadership - previous version of this optimization was that we didn't even tell the leader that it's a leader to avoid doing assignment, which resulted in the problem partially fixed by this change).

If it has other side effects, let's add a comment that explains that.

Reply:
The fundamental issue is that the assignor is not idempotent nor free of any side effects (e.g. custom ones) so running it and ignoring the new assignment would lead to a weird state for the group. This is briefly discussed in the KIP, see rejected alternatives. I have updated the comment in the GroupCoordinator to stress out this point. Does this work for you?

Reply:
Sounds good.